nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/io/avro/AvroMapperRepositoryInit.java

```diff
@@ -0,0 +1,104 @@
+        DocumentModel doc = session.createDocumentModel("/", name, "ComplexDoc");
+        // common
+        doc.getProperty("common:icon").setValue("/path/to/my/icon");
+        if (full) {
+            doc.getProperty("common:icon-expanded").setValue("/path/to/my/icon-expanded");
+        }
+        // dublin
+        doc.getProperty("dublincore:modified").setValue(new Date());
+        doc.getProperty("dublincore:created").setValue(new Date());
+        doc.getProperty("dublincore:title").setValue("title");
+        doc.getProperty("dublincore:rights").setValue("rights");
+        doc.getProperty("dublincore:subjects").setValue(new String[] { "subject1", "subject2", "subject3" });
+        if (full) {
+            doc.getProperty("dublincore:expired").setValue(new Date());
+            doc.getProperty("dublincore:issued").setValue(new Date());
+            doc.getProperty("dublincore:valid").setValue(new Date());
+            doc.getProperty("dublincore:nature").setValue("nature");
+            doc.getProperty("dublincore:source").setValue("source");
+            doc.getProperty("dublincore:format").setValue("format");
+            doc.getProperty("dublincore:creator").setValue("creator");
+            doc.getProperty("dublincore:coverage").setValue("coverage");
+            doc.getProperty("dublincore:language").setValue("language");
+            doc.getProperty("dublincore:publisher").setValue("publisher");
+            doc.getProperty("dublincore:description").setValue("description");
+            doc.getProperty("dublincore:lastContributor").setValue("lastContributor");
+            doc.getProperty("dublincore:contributors").setValue(new String[] { "contrib1", "contrib2", "contrib3" });
+        }
+        // complexschema
+        if (full) {
+            doc.getProperty("complexschema:attachedFile").setValue("name", "attachedFileName");
+        }
+        ListProperty vignettes = (ListProperty) doc.getProperty("complexschema:attachedFile/vignettes");
+        for (String label : new String[] { "toto", "titi", "tutu", "tata" }) {
+            vignettes.addValue(createVignette(label, full));
+        }
+        session.createDocument(doc);
+    }
+
+    private Map<String, Object> createVignette(String label, boolean full) {
+        try {
+            Blob blob = full
+                    ? Blobs.createBlob(label.getBytes("UTF-8"), "plain/test", "UTF-8")
+                    : Blobs.createBlob(label);
+            blob.setDigest("gest");
+            Map<String, Object> vignette = new HashMap<>(4);
+            if (full) {
+                vignette.put("label", label);
+                vignette.put("width", Integer.valueOf(new Random().nextInt(1000)));
+            }
+            vignette.put("height", Integer.valueOf(new Random().nextInt(1000)));
+            vignette.put("content", blob);
+            return vignette;
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
As it's test it's better to not catch the exception. You can add throw Exception on your method signature, or wrap the `IOException` in a runtime one.